test/client_test.exs

```diff
@@ -39,18 +39,74 @@ defmodule Sentry.ClientTest do
 
     Application.put_env(:sentry, :dsn, "http://public:secret@localhost:#{bypass.port}/1")
 
-      try do
-        Event.not_a_function
-      rescue
-        e ->
+    try do
+      Event.not_a_function
+    rescue
+      e ->
         assert capture_log(fn ->
           Sentry.capture_exception(e)
         end) =~ ~r/400.*Creation of this event was denied due to rate limiting/
-      end
+    end
   end
 
   test "errors when attempting to report invalid JSON" do
     unencodable_tuple = {:a, :b, :c}
     assert :error = Sentry.capture_message(unencodable_tuple)
   end
+
+  test "calls anonymous pre_event_send_function" do
+    bypass = Bypass.open
+    Bypass.expect bypass, fn conn ->
+      {:ok, body, conn} = Plug.Conn.read_body(conn)
+      request_map = Poison.decode!(body)
+      assert request_map["extra"] == %{"key" => "value"}
+      assert request_map["user"]["id"] == 1
+      Plug.Conn.resp(conn, 200, ~s<{"id": "340"}>)
+    end
+
+    Application.put_env(:sentry, :dsn, "http://public:secret@localhost:#{bypass.port}/1")
+    Logger.metadata([key: "value", user_id: 1])
+
+    Application.put_env(:sentry, :pre_event_send_function, fn(e) ->
+      metadata = Enum.into(Logger.metadata, %{})
```

Comment:
This can be simplified to `Map.new(Logger.metadata)` (which is also faster than `Enum.into`).